drivers/soundwire/intel.c

```diff
@@ -1575,6 +1576,32 @@ static int intel_suspend(struct device *dev)
 		 * we will need to reset the pm_runtime status to active
 		 */
 		md->pm_runtime_suspended = true;
+		clock_stop_quirks = sdw->link_res->clock_stop_quirks;
+
+		if ((clock_stop_quirks & SDW_INTEL_CLK_STOP_BUS_RESET ||
+		     !clock_stop_quirks) &&
+		    !pm_runtime_suspended(dev->parent)) {
+
+			/*
+			 * if we've enabled clock stop, and the parent
+			 * is still active, disable shim wake. The
+			 * SHIM registers are not accessible if the
+			 * parent is already pm_runtime suspended so
+			 * it's too late to change that
+			 * configuration
+			 */
+
+			dev_dbg(dev, "%s: disabling shim wake\n", __func__);
+
+			intel_shim_wake(sdw, false);
+
+			/*
+			 * FIXME: if we suspend with the clock stop
+			 * mode turned on, can we resume without
+			 * issues, or do we need to turn-off the
+			 * clock-stop mode before suspending?
+			 */
```

Comment:
Currently, we need to reset bus to sync with slave after resume. Sdw driver is stable in S3 test. And it is hard to make S3 test failed according to QA's test. I discussed with @shuming and he didn't imagine any risk